### DISCUSSION TITLE
argo-cd continues to call API deprecated in Kubernetes v1.22

### DISCUSSION THREAD

**Discussion by tatsuro-m** [argoproj/argo-cd#9899] [Q&A]
### Version, etc.
- Kubernetes 1.21
- ArgoCD 2.1.15
- Kubernetes environment: GKE (GCP)

### Goal
Safely update Kubernetes version to 1.22.

---

Hello.
Can someone please help me 🙏

I have been trying to get my k8s version up to 1.22 and have updated the API that will no longer be available with k8s 1.22.
https://cloud.google.com/kubernetes-engine/docs/deprecations/apis-1-22

Following the documentation, we have addressed the following two APIs that will no longer be available
- `/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions`
- `/apis/extensions/v1beta1/ingresses`

I thought there would be no more deprecated API calls in the cluster.
However, when I run the following query in Cloud Logging, `/apis/extensions/v1beta1/ingresses` is still being called. 😭

```
resource.type="k8s_cluster"
labels. "k8s.io/removed-release"="1.22"
protoPayload.authenticationInfo.principalEmail:("system:serviceaccount" OR "@")
protoPayload.authenticationInfo.principalEmail!~("system:serviceaccount:kube-system:")
```
```
{
  "protoPayload": {
    "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
    "authenticationInfo": {
      "principalEmail": "system:serviceaccount:argocd:argocd-application-controller"
    },
    "authorizationInfo": [
      {
        "granted": true,
        "permission": "io.k8s.extensions.v1beta1.ingresses.watch",
        "resource": "extensions/v1beta1/ingresses"
      }
    ],
    "methodName": "io.k8s.extensions.v1beta1.ingresses.watch",
   
```

The log shows that every 10 minutes `argocd-application-controller` calls `/apis/extensions/v1beta1/ingresses` which is not present in the cluster.
The only client calling the deprecated API is `argocd-application-controller`.

## Question
1. Why is this happening?
2. In such a case, we think it is ok to update k8s, because it does not matter if `argocd-applicaiton-controller` can no longer call `/v1beta1/ingresses` API in k8s v1.23. Is this idea wrong?

**Comment by ashutosh16** [comment]
As long as none of your applications manages a deprecated resource, you can safely upgrade your cluster version and the messages will disappear (because the API is not available anymore).

Refer to the issue for detail  information: https://github.com/argoproj/argo-cd/issues/9786
